<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>732</width>
    <height>866</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>视频合并工具</string>
  </property>
  <property name="toolTipDuration">
   <number>-1</number>
  </property>
  <property name="styleSheet">
   <string notr="true">QComboBox { color: rgb(0, 0, 0);
	alternate-background-color: rgb(255, 255, 255);
}
</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_12">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>必填</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>选择一个包含视频文件夹或者一个包含视频路径的txt文件(允许拖拽文件进入下方输入框)</string>
          </property>
          <property name="buddy">
           <cstring>lineEdit</cstring>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="FileDragAndDropLineEdit" name="lineEdit">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;推荐使用导入txt的方式进行合并，能保证顺序不会出现错乱&lt;/p&gt;&lt;p&gt;通过点击下面的【文件排序工具】对视频进行排序，然后生成txt排序文件&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="toolTipDuration">
             <number>-1</number>
            </property>
            <property name="dragEnabled">
             <bool>true</bool>
            </property>
            <property name="placeholderText">
             <string>输入文件的路径</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="radioButton_2">
            <property name="text">
             <string>选择txt文件</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="radioButton">
            <property name="text">
             <string>选择文件夹</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_2">
          <property name="text">
           <string>选择视频文件夹/txt文件</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QFrame" name="frame">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;设置搜索视频的方式(遵循fnmatch格式)&lt;/p&gt;&lt;p&gt;该选项仅在使用文件夹合并视频的时候才会生效&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="buddy">
            <cstring>lineEdit_2</cstring>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_2">
           <property name="placeholderText">
            <string>*.mp4</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>更多设置</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <widget class="QScrollArea" name="scrollArea">
        <property name="frameShape">
         <enum>QFrame::NoFrame</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Sunken</enum>
        </property>
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="scrollAreaWidgetContents">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>694</width>
           <height>265</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_10">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <widget class="QLabel" name="label_5">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>输出文件路径(允许拖拽文件进入下方输入框)</string>
              </property>
              <property name="buddy">
               <cstring>lineEdit_3</cstring>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="FileDragAndDropLineEdit" name="lineEdit_3">
                <property name="text">
                 <string>output.mp4</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_3">
                <property name="text">
                 <string>选择路径</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_6">
              <item>
               <widget class="QLabel" name="label_6">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;输出的视频将会被&lt;span style=&quot; font-weight:700;&quot;&gt;平滑抽帧&lt;/span&gt;或者&lt;span style=&quot; font-weight:700;&quot;&gt;平滑补帧&lt;/span&gt;到这个帧数&lt;/p&gt;&lt;p&gt;相比传统OpenCv的抽帧和补帧，该软件采用特殊的平滑抽帧和补帧从全局进行补帧或者抽帧，能&lt;span style=&quot; font-weight:700;&quot;&gt;最大程度的保证视频流畅和连贯&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
                <property name="text">
                 <string>输出视频帧率</string>
                </property>
                <property name="buddy">
                 <cstring>spinBox</cstring>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="spinBox">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;输出的视频将会被&lt;span style=&quot; font-weight:700;&quot;&gt;平滑抽帧&lt;/span&gt;或者&lt;span style=&quot; font-weight:700;&quot;&gt;平滑补帧&lt;/span&gt;到这个帧数&lt;/p&gt;&lt;p&gt;相比传统OpenCv的抽帧和补帧，该软件采用特殊的平滑抽帧和补帧从全局进行补帧或者抽帧，能&lt;span style=&quot; font-weight:700;&quot;&gt;最大程度的保证视频流畅和连贯&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
                <property name="maximum">
                 <number>200</number>
                </property>
                <property name="value">
                 <number>25</number>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_7">
              <item>
               <widget class="QLabel" name="label_7">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;软件采用自研去黑边算法，自动识别并去除边角Logo，字幕，动态图标等，能精准获取主体画面&lt;/span&gt;，相比传统算法提取主体画面的速度&lt;span style=&quot; font-weight:700;&quot;&gt;提高5倍以上&lt;/span&gt;(霍夫,非极大值抑制)&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;该取值是从画面中进行采样的比例,0表示完全关闭去黑边功能,1表示去黑边功能最大化,适合您的视频较暗或者边缘未能完全去除的情况，数值越大去黑边效果越好，速度越慢&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
                <property name="text">
                 <string>视频去黑边采样率</string>
                </property>
                <property name="buddy">
                 <cstring>doubleSpinBox</cstring>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="doubleSpinBox">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;软件采用自研去黑边算法，自动识别并去除边角Logo，字幕，动态图标等，能精准获取主体画面&lt;/span&gt;，相比传统算法提取主体画面的速度&lt;span style=&quot; font-weight:700;&quot;&gt;提高5倍以上&lt;/span&gt;(霍夫,非极大值抑制)&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;该取值是从画面中进行采样的比例,0表示完全关闭去黑边功能,1表示去黑边功能最大化,适合您的视频较暗或者边缘未能完全去除的情况，数值越大去黑边效果越好，速度越慢&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
                <property name="maximum">
                 <double>1.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.050000000000000</double>
                </property>
                <property name="value">
                 <double>0.500000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_7">
            <item>
             <widget class="QLabel" name="label_8">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="toolTip">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;选择输出视频的方向,软件会自动将所有的视频转换成该方向&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="text">
               <string>输出视频方向</string>
              </property>
              <property name="buddy">
               <cstring>comboBox</cstring>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="comboBox">
              <property name="toolTip">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;选择输出视频的方向,软件会自动将所有的视频转换成该方向&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="currentIndex">
               <number>-1</number>
              </property>
              <property name="placeholderText">
               <string/>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_9">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_8">
              <item>
               <widget class="QLabel" name="label_9">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="toolTip">
                 <string>如果您选择了竖屏，那么请选择该选项让程序知道您要处理您的横屏视频
如果您的横屏视频只需要顺时针旋转90°就能变成竖屏则在此处选择顺时针旋转90°
注意，如果您旋转出错您的画面会出现更大的黑边</string>
                </property>
                <property name="text">
                 <string>横屏视频旋转方向</string>
                </property>
                <property name="buddy">
                 <cstring>comboBox_2</cstring>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="comboBox_2">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;如果您选择了竖屏，那么请选择该选项让程序知道您要处理您的横屏视频&lt;/p&gt;&lt;p&gt;如果您的横屏视频只需要顺时针旋转90°就能变成竖屏则在此处选择顺时针旋转90°&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;注意，如果您旋转出错您的画面会出现更大的黑边&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
                <property name="placeholderText">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_9">
              <item>
               <widget class="QLabel" name="label_10">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="toolTip">
                 <string>如果您选择了横屏，那么请选择该选项让程序知道您要处理您的竖屏视频
如果您的竖屏视频只需要顺时针旋转90°就能变成横屏则在此处选择顺时针旋转90°
注意，如果您旋转出错您的画面会出现更大的黑边</string>
                </property>
                <property name="text">
                 <string>竖直视频旋转方向</string>
                </property>
                <property name="buddy">
                 <cstring>comboBox_3</cstring>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="comboBox_3">
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;如果您选择了横屏，那么请选择该选项让程序知道您要处理您的竖屏视频&lt;/p&gt;&lt;p&gt;如果您的竖屏视频只需要顺时针旋转90°就能变成横屏则在此处选择顺时针旋转90°&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;注意，如果您旋转出错您的画面会出现更大的黑边&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_4">
     <property name="title">
      <string>进度条</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>获取信息</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QProgressBar" name="progressBar">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>1</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
          <property name="textVisible">
           <bool>true</bool>
          </property>
          <property name="format">
           <string>(%v/%m) %p%</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>详细进度</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QProgressBar" name="progressBar_2">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>1</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
          <property name="format">
           <string>(%v/%m) %p%</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QPushButton" name="pushButton">
       <property name="font">
        <font>
         <pointsize>17</pointsize>
        </font>
       </property>
       <property name="text">
        <string>开始合并</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCommandLinkButton" name="commandLinkButton">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>进入文件排序工具</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>输出</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_11">
      <item>
       <widget class="CMDTextEdit" name="widget" native="true">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>150</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>FileDragAndDropLineEdit</class>
   <extends>QLineEdit</extends>
   <header>src.components.file_drag_and_drop_lineedit</header>
  </customwidget>
  <customwidget>
   <class>CMDTextEdit</class>
   <extends>QWidget</extends>
   <header>src.components.cmd_text_edit</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>lineEdit</tabstop>
  <tabstop>radioButton_2</tabstop>
  <tabstop>radioButton</tabstop>
  <tabstop>pushButton_2</tabstop>
  <tabstop>lineEdit_2</tabstop>
  <tabstop>scrollArea</tabstop>
  <tabstop>lineEdit_3</tabstop>
  <tabstop>pushButton_3</tabstop>
  <tabstop>spinBox</tabstop>
  <tabstop>doubleSpinBox</tabstop>
  <tabstop>comboBox</tabstop>
  <tabstop>comboBox_2</tabstop>
  <tabstop>comboBox_3</tabstop>
  <tabstop>pushButton</tabstop>
  <tabstop>commandLinkButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>radioButton_2</sender>
   <signal>clicked(bool)</signal>
   <receiver>frame</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>801</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>700</x>
     <y>130</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton</sender>
   <signal>clicked(bool)</signal>
   <receiver>frame</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>874</x>
     <y>71</y>
    </hint>
    <hint type="destinationlabel">
     <x>757</x>
     <y>131</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
